### ISSUE TITLE
BulkDescriptor DeleteMany use IEnumerable<Id>

### ISSUE THREAD

**Issue by russcam** [elastic/elasticsearch-net#4125]
Relates: #3500, #4120 

From the recent conversation in #4120, I propose that we look at adding an overload to bulk `DeleteMany` that accepts an `IEnumerable<Id>`, and remove the overloads that accept `IEnumerable<string>` and `IEnumerable<long>` for 8.x. This would require a user to specify a collecton as `IEnumerable<Id>`, which with implicit conversions from `string` and `long`, would look like

```c#
var ids = new Id[] 
{
	"1",
	"2",
	"3",
};

var client = new ElasticClient();

var bulkResponse = client.Bulk(b => b
	.DeleteMany<object>(ids)
);
```

**Comment by antja0** [comment]
I'm sorry I said I will be doing this and this is still on my work list but it's been a year and I just don't seem to have time to develop this. (consumed by work and I think situation will just keep getting worse 😄).

If I remember correctly this would be fairly straight forward for anyone with strong understanding of c# even if new to the project, so I would recommend to label this [![label: Low hanging fruit][~Low hanging fruit]](https://github.com/elastic/elasticsearch-net/labels/Low%20hanging%20fruit) or similar

[~Low hanging fruit]: https://img.shields.io/badge/-Low%20hanging%20fruit-ff4500.svg

**Comment by stevejgordon** [comment]
I've addressed the API design for the v8.0.0 with some further changes which will appear in the next alpha. I'm going to close this on that basis.
